### ISSUE TITLE
if same gsp resource is in 2 different bundels then it oonly gets added to one bundle

### ISSUE THREAD

**Issue by kapitall** [mckenfra/gsp-resources#9]
If I have the same gsp resource in 2 different bundles then the gsp resource only gets added to the last defined bundle

example if ApplicationResources is like this

```
 bundle1 {
   resource url:'js/libs/path.js'
   resource url:'js/gsp-resource/dna.js.gsp'
 }

  bundle2 {
     resource url:'js/libs/path.js'
     resource url:'js/gsp-resource/dna.js.gsp'
   }
```

then resource dna.js.gsp only gets added to bundle 2 and bundle1 only has path.js in it

Thanks --Serge

**Comment by yenchieh** [comment]
I have the same issue, Can someone help or give us an ETA on resolution.
Cheers
Jay

**Comment by mckenfra** [comment]
Hi,

Thanks for raising this. The problem is caused by an optimisation in the `gsp-resources` plugin, which is to stop the same `.gsp` file being rendered more than once. A quick and simple solution would be to just treat multiple declarations of the same `.gsp` file as separate resources. This would then mean that the `.gsp` file will get re-rendered for every bundle it appears in, allowing the `.gsp` output to be included in multiple bundles.

This is fine if the logic in the `.gsp` is fairly "lightweight". However, if the `.gsp` is doing something slow like hitting the database, then doing the same thing over and over for every module the `.gsp` appears in could slow down server startup.

I think an ideal solution would be to treat each declaration of the same `.gsp` as a separate resource, but to add some new logic before rendering to check the date of the rendered (javascript/css) file. If it is older then the date of the original `.gsp` file, then continue with rendering. But if it is the same date as the `.gsp` file (or newer), then skip rendering and just use the existing rendered file.

One possible downside of this is that currently the `gsp-resources` plugin always re-renders all `.gsp` files on server startup, regardless of whether or not they have been modified. This fix would mean it would only ever re-render a `.gsp` file if it has been modified, otherwise it will just use the previously-rendered file.

Suggest I roll out a temporary snapshot version that just blindly always re-renders. Then I'll work on a more robust solution that prevents rendering if the rendered file already exists and is the same date or newer. Think I'll also need to add a config setting to allow forcing re-rendering on startup in case people still need that.

I'll let you know when the 'simple solution' snapshot is ready.

Cheers,
Francis

**Comment by mckenfra** [comment]
As per my previous comment, I have released the 'simple solution' as version `0.5-SNAPSHOT`. You will need to add the following to your `BuildConfig.groovy`:

```
plugins {
    // Other plugins
    // ...

    runtime ':gsp-resources:0.5-SNAPSHOT'
}
```

Let me know if if works ok.

Cheers,
Francis

**Comment by skreiker** [comment]
Thanks Francis,
I tested and this is now working for both js and css.
Thanks --Serge
